kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/AesGcmIvGenerator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.dek;
+
+import java.security.SecureRandom;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import javax.security.auth.Destroyable;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * RBG-based construction of the Initialization Vector for AES-GCM.
+ * The random field is 96 bits and the free field is empty.
+ *
+ * @see "§8.2.2 of NIST SP.800-38D: Recommendation for Block
+ * Cipher Modes of Operation:
+ * Galois/Counter Mode (GCM)
+ * and GMAC"
+ */
+@NotThreadSafe
+class AesGcmIvGenerator implements Destroyable {
```

Comment:
This might be better named `Wrapping96BitCounter`, since that's what it actually does.